### PR TITLE
Rename Pedigree to RelationshipGraph

### PR DESCRIPTION
**Pull request by stevenhwu**
- This is related to issues #85. The main purpose of this PR is renaming pedigree class to RelationshipGraph. I try to minimize the change in this PR, so other files are only updated as needed.
- Naming cleanup required at the next PR. e.g. variable `Pedigree pedigree_;` in the old file become `RelationshipGraph pedigree_;`. At some stage all these should be change to `RelationshipGraph relationship_graph_;`.
- Add unittest for relationship_graph2, which can be merged with relationship_graph later.
- Next PR will restructure RelationshipGraph a bit, and more unittest for this class.
- Note: dng::io::Pedigree remain unchanged.
- The following 3 files show the difference between `pedigree.*` and `relationship_graph.*`. Generate by

```
git diff develop:Tests/Unit/dng/pedigree.cc feature/RelationshipGraph_class_unittest_PR2:Tests/Unit/dng/relationship_graph.cc
git diff develop:src/include/dng/pedigree.h feature/RelationshipGraph_class_unittest_PR2:src/include/dng/relationship_graph.h
git diff develop:src/lib/pedigree.cc feature/RelationshipGraph_class_unittest_PR2:src/lib/relationship_graph.cc 
```

[gitDiffPedigree.cc.txt](https://github.com/denovogear/denovogear/files/460496/gitDiffPedigree.cc.txt)
[gitDiffPedigree.h.txt](https://github.com/denovogear/denovogear/files/460497/gitDiffPedigree.h.txt)
[gitDiffPedigreeUnittest.txt](https://github.com/denovogear/denovogear/files/460498/gitDiffPedigreeUnittest.txt)

### CODE REVIEW

**Comment by reedacartwright** [comment]
Jenkins, retest this please

**Comment by reedacartwright** [comment]
Jenkins, retest this please

**Comment by reedacartwright** [comment]
Jenkins, retest this please

**Comment by stevenhwu** [comment]
Do you want me to include the updated coverage report and retest it?

**Comment by reedacartwright** [comment]
Jenkins, retest this please

**Comment by reedacartwright** [comment]
Jenkins, retest this please

**Comment by reedacartwright** [comment]
Jenkins, okay to test

@stevenhwu, yes you can update your PR. I'm just using it to test the jenkins server right now.

**Comment by reedacartwright** [comment]
Jenkins, okay to test

**Comment by reedacartwright** [comment]
jenkins, okay to test

File: src/lib/relationship_graph.cc [hunk 1]
```diff
@@ -17,7 +18,7 @@
  * this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <dng/pedigree.h>
+#include "../include/dng/relationship_graph.h"
```
Comment (reedacartwright) [inline src/lib/relationship_graph.cc]: Don't use relative syntax here.

File: src/include/dng/relationship_graph.h [hunk 2]
```diff
@@ -128,9 +129,9 @@ class Pedigree {
     std::vector<transition_t> transitions_;
 
     // The original, simplified peeling operations
-    std::vector<decltype(peel::op::NUM)> peeling_ops_;
+    std::vector<decltype(peel::op::OpTypes::NUM)> peeling_ops_;
     // The modified, "faster" operations
```
Comment (reedacartwright) [inline src/include/dng/relationship_graph.h:133]: Why are we using OpTypes:: here ?

File: src/include/dng/peeling.h [hunk 3]
```diff
@@ -35,7 +35,7 @@ namespace dng {
 namespace peel {
 
 namespace op {
-enum {
+enum OpTypes {
     UP, DOWN, TOFATHER, TOMOTHER, TOCHILD,
```
Comment (reedacartwright) [inline src/include/dng/peeling.h:39]: Is this change only to silence a warning?

Comment (stevenhwu) [reply src/include/dng/peeling.h:39]: Yes, only purpose is to get ride off 
`warning: ‘dng::RelationshipGraph’ has a field ‘dng::RelationshipGraph::peeling_ops_’ whose type uses the anonymous namespace [-Wsubobject-linkage]`

Comment (reedacartwright) [reply src/include/dng/peeling.h:39]: Since we are already in namespace op, lets change it to 'Type' from 'OpTypes'

**Comment by reedacartwright** [comment]
Jenkins, retest this please

**Comment by reedacartwright** [comment]
Jenkins, retest this please.

**Pull request merged**